src/iidm/converter/xml/IidmXmlUtil.cpp

```diff
@@ -21,38 +26,68 @@ namespace converter {
-    return PowsyblException(stdcxx::format("%1%.%2% is %3% for IIDM-XML version %4%. %5%%6%", rootElementName, elementName, errorMessage, contextVersion.toString("."), reason, version.toString(".")));
+    return createException(stdcxx::format("%1%.%2%", rootElementName, elementName), errorMessage, version, contextVersion, reason);
 }
 
-void IidmXmlUtil::assertMaximumVersion(const std::string& rootElementName, const std::string& elementName, const char* errorMessage, const IidmXmlVersion& maxVersion, const IidmXmlVersion& contextVersion) {
-    if (contextVersion > maxVersion) {
-        throw createException(rootElementName, elementName, errorMessage, maxVersion, contextVersion, "IIDM-XML version should be <= ");
+void createExceptionOrLogError(const std::string& elementName, const char* errorMessage, const IidmXmlVersion& refVersion, const std::string& reason, const NetworkXmlWriterContext& context) {
+    static logging::Logger& S_LOGGER = logging::LoggerFactory::getLogger<IidmXmlUtil>();
+    if (context.getOptions().getIidmVersionIncompatibilityBehavior() == ExportOptions::IidmVersionIncompatibilityBehavior::THROW_EXCEPTION) {
+        throw createException(elementName, errorMessage, refVersion, context.getVersion(), reason);
     }
+    if (context.getOptions().getIidmVersionIncompatibilityBehavior() == ExportOptions::IidmVersionIncompatibilityBehavior::LOG_ERROR) {
+        S_LOGGER.error(message(elementName, errorMessage, refVersion, context.getVersion(), reason));
+    } else {
+        throw AssertionError(stdcxx::format("Unexpected behaviour: %1%", context.getOptions().getIidmVersionIncompatibilityBehavior()));
+    }
+}
+
+void createExceptionOrLogError(const std::string& rootElementName, const std::string& elementName, const char* errorMessage, const IidmXmlVersion& refVersion, const std::string& reason, const NetworkXmlWriterContext& context) {
+    createExceptionOrLogError(stdcxx::format("%1%.%2%", rootElementName, elementName), errorMessage, refVersion, reason, context);
 }
 
-void IidmXmlUtil::assertMinimumVersion(const std::string& rootElementName, const std::string& elementName, const char* errorMessage, const IidmXmlVersion& minVersion, const IidmXmlVersion& contextVersion) {
-    if (contextVersion < minVersion) {
-        throw createException(rootElementName, elementName, errorMessage, minVersion, contextVersion, "IIDM-XML version should be >= ");
+void IidmXmlUtil::assertMaximumVersion(const std::string& rootElementName, const std::string& elementName, const char* errorMessage, const IidmXmlVersion& maxVersion, const NetworkXmlWriterContext& context) {
+    if (context.getVersion() > maxVersion) {
+        createExceptionOrLogError(rootElementName, elementName, errorMessage, maxVersion, MAXIMUM_REASON, context);
     }
 }
```

Comment:
In Java version we have two versions of this method: one for the reader and one for the writer